src/atmega328p/hal/gating.rs

```diff
@@ -0,0 +1,107 @@
+// RustDuino : A generic HAL implementation for Arduino Boards in Rust
+// Copyright (C) 2021  Sanmati Pande, Indian Institute of Technology Kanpur
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>
+
+/// Power reduction for ATmega328p chip
+pub enum Peripherals {
+    TWI,
+    Timer2,
+    Timer0,
+    Timer1,
+    SPI, //serial peripheral interface
+    USART0,
+    ADC,
+}
+///registers controlling power management
+/// Section 9.11 of ATmega328p Datasheet
+pub struct Power {
+    //power reduction register control bits for power management.
+    prr: u8,
+}
+///returns mutable reference to Power struct
+impl Power {
```

Comment:
Again, you don't need to write what is obvious in comments like new will return a struct always.